recipes/ftjam/all/test_package/conanfile.py

```diff
@@ -6,22 +6,26 @@
 class TestPackage(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
 
+    def build_requirements(self):
+        if hasattr(self, "settings_build"):
+            self.build_requires(str(self.requires["ftjam"]))
```

Comment:
This causes a bootstrap problem on c3i when adding a new recipe.
I'm not sure what version of the build requirement is going to be used.

See https://github.com/conan-io/conan-center-index/pull/6677#issuecomment-893434563

Reply:
Well, the issue here is that we don't want to test ftjam in host context, but in build context. This trick works as soon as `foo` in `self.build_requires(str(self.requires["foo"]))` is the tested recipe, and not a dependency of this recipe.

Reply:
So we're fine that we're building `ftjam` for the host machine, but are actually testing `ftjam` for the build machine?

Reply:
Ahah yes, but cross-build of ftjam doesn't work anyway. Actually here I was able to see this issue while using 2 same profiles.